src/internal/hooks/useMessage.ts

```diff
@@ -216,3 +226,67 @@ const statusSetter = (setStatus: Dispatch<SetStateAction<MessageStatus>>) => {
     }
   };
 };
+
+export const useMessageSenders = () => {
```

Comment:
Not sure if there's a better name, but nice to move this boilerplate out of editor.tsx

Reply:
I was thinking of doing something like this. Unfortunately it doesn't work for the receivers too since those need a bunch of props.